src/main/java/de/unistuttgart/informatik/fius/icge/simulation/Simulation.java

```diff
@@ -144,63 +144,96 @@ public void setTerritory(Territory tty) {
+    }
+
+    /**
+     * @return All alive `CollectableEntity`s within this `Simulation`
+     */
+    public ArrayList<CollectableEntity> collectables() {
+        return this.collectablesWith(ent -> true);
     }
 
     /**
-     * @return All collectables in this simulation
+     * Gets all alive `CollectableEntity`s within this `Simulation` that match a certain predicate
+     * 
+     * @param pred
+     *            The predicate that the `CollectableEntity`s are tested for
+     * @return The matching `CollectableEntity`s
      */
-    public synchronized ArrayList<CollectableEntity> collectables() {
+    public synchronized ArrayList<CollectableEntity> collectablesWith(Predicate<CollectableEntity> pred) {
         ArrayList<CollectableEntity> result = new ArrayList<>();
-        this._entityObjects.keySet().forEach(ent -> {
-            WorldObject wob = this.worldObject(ent);
-            if ((ent instanceof CollectableEntity)) {
-                result.add((CollectableEntity) ent);
-            }
-        });
+        this._entityObjects.keySet().stream().filter(ent -> ent instanceof CollectableEntity)
+                .map(ent -> (CollectableEntity) ent).filter(pred).forEach(result::add);
         return result;
     }
 
     /**
-     * Get all collectables in the given row and column
-     * 
-     * @param row
-     *            The row the collectables must be in
+     * Gets all alive `CollectableEntity`s within this `Simulation` that are in a certain cell
      * 
      * @param column
-     *            The column the collectables must be in
-     * @return all collectable entities in given cell
+     *            The column of the cell
+     * @param row
+     *            The row of the cell
+     * @return The `CollectableEntity`s that are in the specified cell
      */
     public synchronized ArrayList<CollectableEntity> collectablesAt(int column, int row) {
         ArrayList<CollectableEntity> result = new ArrayList<>();
-        this._entityObjects.keySet().forEach(ent -> {
-            WorldObject wob = this.worldObject(ent);
-            if ((ent instanceof CollectableEntity) && (wob.row == row) && (wob.column == column)) {
-                result.add((CollectableEntity) ent);
-            }
-        });
+        this._entityObjects.keySet().stream().filter(ent -> ent instanceof CollectableEntity)
+                .filter(Entity.predicateIsAt(column, row)).map(ent -> (CollectableEntity) ent).forEach(result::add);
         return result;
     }
 
     /**
-     * @param ent
-     *            entity to test
-     * @return true if entity is part of this simulation
+     * Checks if this `Simulation` contains at least on alive `Entity` that matches a certain predicate
+     * 
+     * @param pred
+     *            The predicate that the `Entity`s are tested for
+     * 
+     * @return true iff at least one alive `Entity` matches the specified predicate
      */
-    public boolean contains(Entity ent) {

```

Comment:
Why remove the contains method? It should be used for the entity isAlive method as it should have a better performance than the current alive check.

Reply:
This performance doesn't matter and the `contains()` method might be confusing in the following situation:
* `Entity ent` is not `alive()`
* `ent.simulation() == sim`
* `sim.contains(ent)` returns `false`
You see how that `contains()` would return false even though `ent` is associated with `sim`. This is the right behaviour but might be unexpected.

I'd rather go with writing one of
* `ent.isAlive()` or
* `ent.simulation() == sim`
depending on the desired semantics.